service/controller/key/key.go

```diff
@@ -21,10 +24,36 @@ const (
 	ServiceTypeManaged = "managed"
 )
 
-func ReleaseVersion(customResource releasev1alpha1.Release) string {
-	return customResource.Spec.Version
+// ReleaseAppCRName returns the name of the release App CR for the given release cycle.
+func ReleaseAppCRName(releaseCycleCR releasev1alpha1.ReleaseCycle) string {
```

Comment:
The app CR wording confuses me. The type given is `ReleaseCycle`. What does "app" have to do with it? What app?

Reply:
The idea here is: from a release cycle CR we produce the name of the app CR which is going to install the release CR.

Not sure how to express this better.

Reply:
Is it useful to expose that in the `key` package. I think it should be known only by the `app` resource in `releasecycle` controller.

Reply:
make sense. I moved the function to the `app` resource.